synapse/app/synctl.py

```diff
@@ -190,7 +199,11 @@ def main():
         if start_stop_synapse:
             stop(pidfile, "synapse.app.homeserver")
 
-        # TODO: Wait for synapse to actually shutdown before starting it again
+    # Wait for synapse to actually shutdown before starting it again
+    if action == "restart":
+        write("Waiting to restart ...")
+        while os.path.exists(pidfile):
```

Comment:
should we not check if the process is actually running, rather than just checking for the existence of the pidfile?